SWLOR.Game.Server/Feature/GuiDefinition/ViewModel/MusicManagerViewModel.cs

```diff
@@ -0,0 +1,223 @@
+            get => Get<string>();
+            set => Set(value);
+        }
+
+        public bool IsMusicSelected
+        {
+            get => Get<bool>();
+            set => Set(value);
+        }
+
+        public int SelectedMusicIndex
+        {
+            get => Get<int>();
+            set => Set(value);
+        }
+
+        public GuiBindingList<string> MusicList
+        {
+            get => Get<GuiBindingList<string>>();
+            set => Set(value);
+        }
+
+        public GuiBindingList<bool> MusicToggled
+        {
+            get => Get<GuiBindingList<bool>>();
+            set => Set(value);
+        }
+
+        protected override void Initialize(GuiPayloadBase initialPayload)
+        {
+            SearchText = string.Empty;
+            Search();
+            WatchOnClient(model => model.SearchText);
+        }
+
+        public Action OnSelectMusic() => () =>
+        {
+            if (SelectedMusicIndex > -1)
+                MusicToggled[SelectedMusicIndex] = false;
+
+            var index = NuiGetEventArrayIndex();
+            SelectedMusicIndex = index;
+
+            MusicToggled[SelectedMusicIndex] = true;
+            IsMusicSelected = true;
+        };
+
+        public Action OnClickSearch() => Search;
+
+        public Action OnClickClearSearch() => () =>
+        {
+            _songNames.Clear();
+            _songIds.Clear();
+            SelectedPageIndex = 0;
+            SearchText = string.Empty;
+            Search();
+        };
+
+        public Action OnClickPlayMusic() => () =>
+        {
+            var area = GetArea(Player);
+            var musicName = _songNames[SelectedMusicIndex];
+            var musicId = _songIds[SelectedMusicIndex];
+
+            SendMessageToPC(Player, $"You are now playing the song: {musicName}");
+
+            MusicBackgroundChangeDay(area, musicId);
+            MusicBackgroundChangeNight(area, musicId);
+            MusicBackgroundPlay(area);
+        };
+
+        public Action OnClicKStopMusic() => () =>
+        {
+            SendMessageToPC(Player, "Stopped music.");
+
+            MusicBackgroundStop(GetArea(Player));
+        };
+
+        public void Search()
+        {
+            var allSongs = GetAllSongs();
+
+            var pagedallSongs = allSongs
+                .Skip(SelectedPageIndex * ListingsPerPage)
+                .Take(ListingsPerPage)
+                .ToList();
+
+            var musicList = new GuiBindingList<string>();
+            var musicToggled = new GuiBindingList<bool>();
+
+            _songNames.Clear();
+            _songIds.Clear();
+
+            if (string.IsNullOrWhiteSpace(SearchText))
+            {
+                foreach (var song in pagedallSongs)
+                {
+                    var songName = song.DisplayName;
+                    var songID = song.ID;
+
+                    musicList.Add(songName);
+                    _songIds.Add(songID);
+                    _songNames.Add(songName);
```

Comment:
You already have a GUI-bound list for the song name - variable name is MusicList. Why are you storing the name in two places?